src/lib/misc.h

```diff
@@ -32,10 +32,15 @@
 #ifndef MISC_H_
 #define MISC_H_
 
+#include "types.h"
+
 int pgp_keyid(uint8_t *, const size_t, const pgp_pubkey_t *);
 
 int pgp_fingerprint(pgp_fingerprint_t *, const pgp_pubkey_t *);
 
 void pgp_forget(void *, size_t);
 
-#endif
\ No newline at end of file
+int hash_string(pgp_hash_t *hash, const uint8_t *buf, uint32_t len);
```

Comment:
hash.h is probably better place for this

Reply:
Only use of this function is for creating key fingerprints. The PGP fingerprint code is in `misc.c` (where `hash_string` is defined). Only reason it must be declared is it is also used in `key_store_ssh.c` for SSH fingerprints.

Probably best thing is to move `ssh_fingerprint` and `pgp_fingerprint` out of their respective files into some shared `fingerprint.c` and then all of these helper functions can just be static in that one file.

Reply:
ah, ok I see